src/main/kotlin/org/malachite/estella/services/OfferService.kt

```diff
@@ -1,40 +1,68 @@
 import org.malachite.estella.offer.infrastructure.HibernateOfferRepository
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
 import org.springframework.stereotype.Service
+import java.sql.Date
+import java.time.LocalDate
 import java.util.*
+import kotlin.NoSuchElementException
 
 @Service
-class OfferService(@Autowired private val offerRepository: OfferRepository) {
+class OfferService(
+    @Autowired private val offerRepository: OfferRepository,
+    @Autowired private val desiredSkillService: DesiredSkillService,
+    @Autowired private val recruitmentProcessService: RecruitmentProcessService,
+) : EStellaService() {
 
-    fun getOffers(): MutableIterable<Offer> = offerRepository.findByOrderByIdDesc()
+    override val throwable: Exception = OfferNotFoundException()
 
-    fun getOffer(id: Int): Offer = offerRepository.findById(id).get()
+    fun getOffers(): MutableIterable<Offer> =
+        offerRepository.findByOrderByIdDesc()
 
-    fun getOfferDesiredSkills(id: Int): MutableIterable<DesiredSkill> = getOffer(id).skills.toMutableSet()
+    fun getOffer(id: Int): Offer = withExceptionThrower { offerRepository.findById(id).get() } as Offer
 
-    fun addOffer(offer: Offer): Offer = offerRepository.save(offer)
+    fun getOfferDesiredSkills(id: Int): MutableIterable<DesiredSkill> =
+        getOffer(id).skills.toMutableSet()
+
+    fun addOffer(offer: OfferRequest, hrPartner: HrPartner) {
+        val saved: Offer = this.addOffer(offer.toOffer(hrPartner, desiredSkillService))
+        recruitmentProcessService.addBasicProcess(saved)
+    }
+
+    fun addOffer(offer: Offer): Offer = withExceptionThrower { offerRepository.save(offer) } as Offer
 
     fun updateOffer(id: Int, offer: Offer) {
         val currOffer: Optional<Offer> = offerRepository.findById(id)
         if (currOffer.isPresent) {
-            val updated: Offer = currOffer.get().copy(name = offer.name,
-                    description = offer.description,
-                    position = offer.position,
-                    minSalary = offer.minSalary,
-                    maxSalary = offer.maxSalary,
-                    localization = offer.localization,
-                    skills = offer.skills
+            val updated: Offer = currOffer.get().copy(
+                name = offer.name,
+                description = offer.description,
+                position = offer.position,
+                minSalary = offer.minSalary,
+                maxSalary = offer.maxSalary,
+                localization = offer.localization,
+                skills = offer.skills
             )
-
             offerRepository.save(updated)
+        } else {
+            throw OfferNotFoundException()
         }
     }
 
-    fun deleteOffer(id: Int) = offerRepository.deleteById(id)
+    fun updateOffer(id: Int, offer: OfferRequest, hrPartner: HrPartner) {
```

Comment:
Chwilowo wstrzymujemy się z autentykacją tak?

Reply:
No na razie tak, bo nie mamy tego dogadanego, jak chcemy to robić